### PR TITLE
Add link to focus area from add plan page

### PR DESCRIPTION
**Pull request by moshthepitt**
Fix: #649

Adds a link to the affected focus area when adding a new FI plan.

### CODE REVIEW

File: src/components/FormikEffect/index.tsx [hunk 1]
```diff
@@ -0,0 +1,44 @@
+import { connect, FormikContext, FormikState, FormikValues } from 'formik';
+import { useEffect, useRef } from 'react';
+
+/** Type def for Formik onChange function */
+export type FormikOnchangeFunc<T> = (
```
Comment (cKellyDesign) [inline src/components/FormikEffect/index.tsx]: `FormikOnchangeFunc` => `FormikOnChangeFunc` pls

File: src/components/FormikEffect/index.tsx [hunk 2]
```diff
@@ -0,0 +1,44 @@
...
+
+/** Interface to describe FormikEffect props */
+export interface FormikEffectProps<T = FormikValues> {
+  readonly onChange: FormikOnchangeFunc<T>;
+}
+
+/** Interface for connected FormikEffect Props */
+interface ConnectedFormikEffectsProps<T> extends FormikEffectProps<T> {
+  readonly formik: FormikContext<T>;
+}
```
Comment (cKellyDesign) [inline src/components/FormikEffect/index.tsx]: Should this interface also have `onChange`?

File: src/components/FormikEffect/index.tsx [hunk 3]
```diff
@@ -0,0 +1,44 @@
...
+
+/**
+ * Declarative Formik component for side-effects
+ *
```
Comment (cKellyDesign) [inline src/components/FormikEffect/index.tsx]: Should we also include?
```
* @param {ConnectedFormikEffectsProps<T>} props
* @returns null
```

File: src/components/FormikEffect/index.tsx [hunk 4]
```diff
@@ -0,0 +1,44 @@
...
+ * Import the <FormikEffect > component and put it inside any Formik form.
+ * It renders null! Pass it an onChange() function and it will be called
+ * whenever your Formik form's state updates.
+ *
+ * Based on: https://github.com/jaredpalmer/formik-effect
+ * which is currently not working (since React 16.3 API changes)
+ * TODO: We'll need to replace this once formik-effect is updated
+ */
+function FormikEffect<T>(props: ConnectedFormikEffectsProps<T>) {
```
Comment (cKellyDesign) [inline src/components/FormikEffect/index.tsx]: Should this have `: null` as the type being returned?

File: src/containers/pages/InterventionPlan/NewPlan/General/index.tsx [hunk 5]
```diff
@@ -35,8 +43,16 @@ const NewPlan = () => {
...
+        <Col md={8} id="planform-col-container">
+          <PlanForm {...{ formHandler: formValuesHandler }} />
+        </Col>
+        <Col md={4}>
+          {formValues.interventionType === InterventionType.FI &&
+            formValues.jurisdictions[0].id !== '' && (
+              <Link to={`${FI_SINGLE_URL}/${formValues.jurisdictions[0].id}`}>
+                {format(FIS_IN_JURISDICTION, formValues.jurisdictions[0].name)}
+              </Link>
+            )}
```
Comment (cKellyDesign) [inline src/containers/pages/InterventionPlan/NewPlan/General/index.tsx]: 👍

File: src/containers/forms/PlanForm/index.tsx [hunk 6]
```diff
@@ -119,6 +120,10 @@ export interface PlanFormProps {
   cascadingSelect: boolean /** should we use cascading selects for jurisdiction selection */;
   disabledActivityFields: string[] /** activity fields that are disabled */;
   disabledFields: string[] /** fields that are disabled */;
+  formHandler: (
+    curr: any,
+    next: any
+  ) => void /** callback to handle form values, used to pass them to parent components */;
```
Comment (cKellyDesign) [inline src/containers/forms/PlanForm/index.tsx]: Should `curr` and `next` be of type `FormikState<T>`?

File: src/containers/pages/InterventionPlan/NewPlan/General/index.tsx [hunk 7]
```diff
@@ -1,13 +1,21 @@
...
-import PlanForm from '../../../../forms/PlanForm';
+import { CREATE_NEW_PLAN, FIS_IN_JURISDICTION, HOME, PLANS } from '../../../../../configs/lang';
+import { FI_SINGLE_URL, HOME_URL, NEW_PLAN_URL, PLAN_LIST_URL } from '../../../../../constants';
+import { InterventionType } from '../../../../../store/ducks/plans';
+import PlanForm, { defaultInitialValues } from '../../../../forms/PlanForm';
 
 /** Simple component that loads the new plan form and allows you to create a new plan */
 const NewPlan = () => {
+  const [formValues, setFormValues] = useState(defaultInitialValues);
+  const formValuesHandler = (curr: any, next: any) => {
```
Comment (cKellyDesign) [inline src/containers/pages/InterventionPlan/NewPlan/General/index.tsx]: Should `curr` and `next` be of type `FormikState<T>`?

File: src/containers/forms/PlanForm/index.tsx [hunk 8]
```diff
@@ -250,6 +256,7 @@ const PlanForm = (props: PlanFormProps) => {
             }}
             className="mb-5"
           >
+            {formHandler && <FormikEffect onChange={formHandler} />}
```
Comment (cKellyDesign) [inline src/containers/forms/PlanForm/index.tsx]: Nice! Seems like a very unique and very React-ful solution here 👏

**Comment by moshthepitt** [comment]
Thanks for the review @cKellyDesign  - I will implement the changes you identified

**Comment by moshthepitt** [comment]
@cKellyDesign @p-netm this is ready for re-review.

Fair warning: this is a large PR - although it is made up of many smaller ones that you have already reviewed.

407 resolved

File: src/containers/pages/InterventionPlan/NewPlan/General/tests/index.test.tsx [hunk 9]
```diff
@@ -0,0 +1,74 @@
+import { mount, shallow } from 'enzyme';
```
Comment (moshthepitt) [inline src/containers/pages/InterventionPlan/NewPlan/General/tests/index.test.tsx]: This replaces `NewPlan/General/test/index.test.tsx ` which was removed.  (just the test directory was renamed to tests)

File: src/containers/pages/IRS/assignments/index.tsx [hunk 10]
```diff
@@ -155,9 +163,18 @@ interface DispatchedStateProps {
...
+    statusList: planStatus,
+  });
+  const irsPlans = plansArraySelector(state as Registry, {
+    interventionType: InterventionType.IRS,
+    statusList: planStatus,
+  });
+  const mdaPlans = plansArraySelector(state as Registry, {
+    interventionType: InterventionType.MDA,
+    statusList: planStatus,
+  });
```
Comment (cKellyDesign) [inline src/containers/pages/IRS/assignments/index.tsx]: 👍

**Pull request merged**